sdk/samples/iot/templates/iothub/test-resources-post.ps1

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env pwsh
+
+[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $ResourceGroupName,
+
+    [Parameter()]
+    [string] $TestApplicationOid,
+
+    # The DeploymentOutputs parameter is only valid in the test-resources-post.ps1 script.
+    [Parameter()]
+    [hashtable] $DeploymentOutputs,
+
+    # Captures any arguments from eng/New-TestResources.ps1 not declared here (no parameter errors).
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments
+)
+
+###### setup ######
+Install-Module -Name Az -RequiredVersion 4.8.0 -Force -AllowClobber
+Install-Module -Name Az.DeviceProvisioningServices -Force
+
+if ($IsLinux) {
+  $module_location_prefix = "$HOME\.local\share\powershell\Modules"
+  Invoke-Expression -Command "sudo apt install libssl-dev"
+}
+if ($IsWindows) {
+  $module_location_prefix = "$HOME\Documents\PowerShell\Modules"
+}
+if ($IsMacOS) {
+  $module_location_prefix = "$HOME\.local\share\powershell\Modules"
+}
+
+try {
+  Import-Module -Name $module_location_prefix\Az.IotHub -Force
+} catch { throw "Az.IotHub module failed force import" }
+
+try {
+  Import-Module -Name $module_location_prefix\Az.DeviceProvisioningServices -Cmdlet Add-AzIoTDeviceProvisioningServiceLinkedHub -Force 
+} catch { throw "Az.DeviceProvisioningServices module failed force import" }
+
+$orig_loc = Get-Location
+Write-Host $orig_loc
+Write-Host "##vso[task.setvariable variable=VCPKG_ROOT]:$orig_loc/vcpkg"
+cd $orig_loc\sdk\samples\iot\
+$sourcesDir = Get-Location
+
+$region = $DeploymentOutputs['._LOCATION']
+$deviceID = "aziotbld-c-sample"
+$deviceIDSaS = "aziotbld-c-sample-sas"
+$iothubName = $DeploymentOutputs['IOT_HUB_NAME']
+
+###### X509 setup ######
+# Generate certificate 
+openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
+openssl req -new -days 12 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$deviceID"
+
+Get-Content -Path device_ec_cert.pem, device_ec_key.pem | Set-Content -Path device_cert_store.pem
+openssl x509 -noout -fingerprint -in device_ec_cert.pem | % {$_.replace(":", "")} | % {$_.replace("SHA1 Fingerprint=", "")} | Tee-Object -FilePath fingerprint.txt
+$fingerprint = Get-Content -Path .\fingerprint.txt
+
+# Get the hub as an object
+$hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
+
+# Pass fingerprint to IoTHub
+Write-Host "Adding cert device to the allocated hub"
+Add-AzIotHubDevice `
+-InputObject $hub_obj `
+-DeviceId $deviceID `
+-AuthMethod "x509_thumbprint" `
+-PrimaryThumbprint $fingerprint `
+-SecondaryThumbprint $fingerprint
+
+# Download Baltimore Cert
+curl https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem > $sourcesDir\BaltimoreCyberTrustRoot.crt.pem
+
+# sleep, wait for IoTHub device to deploy
+Start-Sleep -s 15
```

Comment:
This introduces a race condition. If possible this should be a loop with a short delay that checks for the presence of the IoTHub device and fails after a while if the device is not present.

Reply:
deleted and it works